### DISCUSSION TITLE
Cannot filter the list using middleware to add or modify e.HttpContext.QueryParam("filter")

### DISCUSSION THREAD

**Discussion by brysalazar12** [pocketbase/pocketbase#1210] [Q&A] [answered]
In client side the collection('mycollection').getList(1, 50, {filter: 'group="1"'}); is working.
But I want to move the filter to middleware

```
func AddFilter(app core.App) echo.MiddlewareFunc {
	return func(next echo.HandlerFunc) echo.HandlerFunc {
		return func(c echo.Context) error {
                         groupId := "1";
			filter := c.QueryParam("filter")

			groupFilter := "group=\"" + groupId + "\""
			if filter != "" {
				groupFilter = filter + " && " + groupFilter
			} 
			c.QueryParams().Set("filter", groupFilter)
			return next(c)
		}
	}
}

func main() {
	app := pocketbase.New()

	app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
		e.Router.Use(AddFilter(app))
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}

```

**Comment by ganigeorgiev** [comment, marked as answer]
This is because internally we are using `c.QueryString()` which is the raw url query and the changes on the on the `c.QueryParams()` doesn't affect it.

In the upcoming v0.9.1 release (most likely tomorrow) I'll consider replacing `c.QueryString()` with `c.QueryParams().Encode()`.

_As a side note, keep in mind that the middleware will execute for each collection requests and this could result in error responses in case the filter contains fields that are not defined in the currently requested records collection, so you may want to add an extra check for `c.Path()`._

**Reply by brysalazar12** [reply]
Thank you
